qemu/tests/qemu_guest_agent.py

```diff
@@ -1310,50 +1311,47 @@ def gagent_check_get_disks(self, test, params, env):
-                          "between guest and qga." % diskname)
-
-            if disk_info_guest['type'] == 'lvm':
-                error_context.context("Check alias", logging.info)
-                cmd_get_disk_alias = params["cmd_get_disk_alias"] % diskname
-                disk_alias = session.cmd_output(cmd_get_disk_alias).strip()
-                if disk_info['alias'] != disk_alias:
-                    test.fail("Disk %s alias is defferent "
-                              "between guest and qga." % diskname)
+        cmd_diskinfo_guest = params['diskinfo_guest_cmd']
+        disks_info_guest = session.cmd_output(cmd_diskinfo_guest)
+        disks_info_guest = json.loads(disks_info_guest)['blockdevices']
+
+        for disk_info_guest in disks_info_guest:
+            diskname = disk_info_guest['kname']
+            error_context.context("Check properties of disk %s"
+                                  % diskname, logging.info)
+            for disk_info_qga in disks_info_qga:
+                diskname_match = False
+                if diskname == disk_info_qga['name']:
+                    diskname_match = True
+                    error_context.context("Check dependencies of disk %s"
+                                          % diskname, logging.info)
+                    if disk_info_guest['type'] == 'disk':
+                        if disk_info_qga['dependencies']:
+                            test.error("Disk %s dependencies "
+                                       "should be null." % diskname)
+                    else:
+                        if disk_info_qga['dependencies'][0] != disk_info_guest['pkname']:
+                            test.fail("Disk %s dependencies is different "
+                                      "between guest and qga." % diskname)
+
+                    error_context.context("Check partition of disk %s"
+                                          % diskname, logging.info)
+                    partition = False if disk_info_guest['type'] != "part" else True
+                    if disk_info_qga["partition"] != partition:
+                        test.fail("Disk %s partition is different "
+                                  "between guest and qga." % diskname)
+
+                    if disk_info_guest['type'] == 'lvm':
+                        error_context.context("Check alias of disk %s"
+                                              % diskname, logging.info)
+                        cmd_get_disk_alias = params["cmd_get_disk_alias"] % diskname
+                        disk_alias = session.cmd_output(cmd_get_disk_alias).strip()
+                        if disk_info_qga['alias'] != disk_alias:
+                            test.fail("Disk %s alias is defferent "
+                                      "between guest and qga." % diskname)
+                    break
+            if not diskname_match:
+                test.fail("Failed to get disk %s with qga." % diskname)
```

Comment:
What about:
```
for disk_info_qga in disks_info_qga:
    if diskname == disk_info_qga['name']:
        ...
        break
else:
    test.fail()
````

Reply:
Make sense, can save one line ~,  will update it.

Reply:
@vivianQizhu  Updated, could you help re-review, thanks!